specs/Standards/NonFungibleToken/Core.md

```diff
@@ -139,33 +139,33 @@ The following behavior is required, but contract authors may name this function
 //
 // The `nft_transfer_call` process:
 //
-// 1. Sender calls `nft_transfer_call` on FT contract
-// 2. NFT contract transfers token from sender to receiver
-// 3. NFT contract calls `nft_on_transfer` on receiver contract
+// 1. Sender calls `nft_transfer_call` on NFT contract.
+// 2. NFT contract transfers token from sender to receiver.
+// 3. NFT contract calls `nft_on_transfer` on receiver contract.
 // 4+. [receiver contract may make other cross-contract calls]
 // N. NFT contract resolves promise chain with `nft_resolve_transfer`, and may
-//    transfer token back to sender
+//    transfer token back to sender.
 //
 // Requirements:
-// * Contract MUST forbid calls to this function by any account except self
-// * If promise chain failed, contract MUST revert token transfer
+// * Contract MUST forbid calls to this function by any account except self.
+// * If promise chain failed, contract MUST revert token transfer.
 // * If promise chain resolves with `true`, contract MUST return token to
-//   `sender_id`
+//   `sender_id`.
 //
 // Arguments:
-// * `sender_id`: the sender of `ft_transfer_call`
-// * `receiver_id`: the `receiver_id` argument given to `ft_transfer_call`
-// * `token_id`: the `token_id` argument given to `ft_transfer_call`
-// * `approved_token_ids`: if using Approval Management, contract MUST provide
-//   set of original approved accounts in this argument, and restore these
+// * `owner_id`: the owner of `token_id`.
+// * `receiver_id`: the `receiver_id` argument given to `nft_transfer_call`.
+// * `token_id`: the `token_id` argument given to `nft_transfer_call`.
+// * `approvals`: if using Approval Management, contract MUST provide.
+//   set of original `approvals` in this argument, and restore these
 //   approved accounts in case of revert.
 //
 // Returns true if token was successfully transferred to `receiver_id`.
 function nft_resolve_transfer(
   owner_id: string,
   receiver_id: string,
   token_id: string,
-  approved_account_ids: null|string[],
+  approvals: null|Record<string, number>,
```

Comment:
This has already been [implemented in near-sdk-rs](https://github.com/near/near-sdk-rs/blob/a903f8c44a7be363d960838d92afdb22d1ce8b87/near-contract-standards/src/non_fungible_token/core/resolver.rs#L29-L39), so we should finally merge it to nomicon